cmds/fmap/lib/fmap.go

```diff
@@ -22,15 +22,19 @@ const (
 	FmapAreaReadOnly
 )
 
-type FMap struct {
-	Start     uint64 // not part of the written struct, but useful nevertheless
+type fMapInternal struct {
 	Signature [8]uint8
 	VerMajor  uint8
 	VerMinor  uint8
 	Base      uint64
 	Size      uint32
 	Name      [32]uint8
-	Areas     []FMapArea
+}
+
+type FMap struct {
+	fMapInternal
+	Start uint64
```

Comment:
Why is there here? Also note if you made this struct exactly match the real fmap header you can write it out again  with no effort. I think that might be useful.